src/sys/options.cxx

```diff
@@ -328,11 +328,17 @@ template <> bool Options::as<bool>(const bool& UNUSED(similar_to)) const {
   
   } else if(bout::utils::holds_alternative<std::string>(value)) {
     auto strvalue = bout::utils::get<std::string>(value);
+
+    // case-insensitve check, so convert string to lower case
+    for (auto& c : strvalue) {
+      c = std::tolower(c);
+    }
   
-    auto c = static_cast<char>(toupper((strvalue)[0]));
-    if ((c == 'Y') || (c == 'T') || (c == '1')) {
+    if ((strvalue == "y") or (strvalue == "yes") or (strvalue == "t")
```

Comment:
Just for my own curiosity, I had a stab at rewriting this to avoid the long-ish chain of comparisons, but I think this is probably worse:

```cpp
auto one_of = [&strvalue](const std::vector<std::string>& values) -> bool {
  return std::any_of(
      values.begin(), values.end(),
      [&strvalue](const auto& value) { return value == strvalue; });
};
if (one_of({"y", "yes", "t", "true", "1"})) {
  result = true;
} else if (one_of({"n", "no", "f", "false", "0"})) {
  result = false;
} else {
```

Could be worth if it `one_of` were a function we could reuse somewhere else.

I think it just betrays my love of lambdas and C++'s love of verbose grammar. Here it is in Python:

```python
if strvalue.lower() in ["y", "yes", "t", "true", "1"]:
  result = True
elif if strvalue.lower() in ["n", "no", "f", "false", "0"]:
   result = False
else:
   raise ValueError()
```